graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -249,4 +250,63 @@ static boolean hasExpectedSchema(Table table) {
     Column contents = table.column(CONTENTS_COLUMN_NAME);
     return contents != null && contents.type() == Column.Type.Varchar;
   }
+
+  public Map<String, Object> tryDeployNewSchema(
+      String namespace, Callable<Map<String, Object>> newSchemaCreator) throws Exception {
+    ensureTableExists(namespace);
+    if (transitionToSchemaDeploymentInProgress(namespace)) {
+      try {
+        return newSchemaCreator.call();
+      } finally {
+        transitionToNoSchemaDeploymentInProgress(namespace);
+      }
```

Comment:
Error handling is going to be a bit tricky: if something fails during the schema validation phase, it's safe to put the flag back to `false`. But if one of the migration queries failed, I think we should leave the flag to `true`: the database is in an inconsistent state, and allowing further migrations on top of it is asking for more trouble. At this point it's probably better if an administrator checks the database and resets the flag manually.

So instead of passing a callable here, I think we should expose the primitives (start deployment, finish deployment, abort deployment), and handle the fine details in the calling code. I'm going to add another comment in `DeploySchemaFetcherBase` to explain how I envision it.